src/components/sorting-options/sorting-options.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+type SortingOptionsProps = {
+  sortingOption: string;
+  onOptionChange: (option: string) => void;
+};
+
+function SortingOptions({sortingOption, onOptionChange}: SortingOptionsProps): JSX.Element {
+  const options = ['Popular', 'Price: low to high', 'Price: high to low', 'Top rated first'];
+  const [optionsIsOpen, setOptionsIsOpen] = useState(false);
+
+  const clickSortOptionsHandle = () => {
+    setOptionsIsOpen(!optionsIsOpen);
+  };
+
+  const clickChooseOptionHandle = (option: string) => {
+    onOptionChange(option);
+    setOptionsIsOpen(!optionsIsOpen);
+  };
```

Comment:
по поводу нейминга, если это название обработчика, то вместо handle должен быть handler, и сначала идёт объект потом действие а потом handler, то есть что то типа 
```suggestion
  const sortFormClickHandler = () => {
    setOptionsIsOpen(!optionsIsOpen);
  };

  const sortOptionClickHandler = (option: string) => {
    onOptionChange(option);
    setOptionsIsOpen(!optionsIsOpen);
  };
```


есть ещё вариант когда функции называют через handle, в ситуация примерно как у тебя сейчас, когда это не обработчик а функция которая внутри обрабтчика вызывается, но тогда тоже лучше немного по другому назвать, clickSortOptionsHandle -> handleSortOptionClick

Reply:
и ещё, в тз об этом не сказано, так что в принципе делать не обязательно, но по хорошему нужно закрывать форму с сортировкой если кликнули где то за её пределами, а то сейчас если города менять она открытой остаётся висеть )